src/System.Diagnostics.Process/tests/ProcessTests.Unix.cs

```diff
@@ -783,6 +783,61 @@ public void TestProcessRecycledPid()
             Assert.True(foundRecycled);
         }
 
+        [Theory]
+        [InlineData(true)]
+        [InlineData(false)]
+        public async Task Kill_ExitedNonChildProcess_DoesNotThrow(bool killTree)
+        {
+            // In this test, we kill a process in a way the Process instance
+            // is not aware the process has terminated when we invoke Process.Kill.
+
+            using (Process nonChildProcess = CreateNonChildProcess())
+            {
+                // Kill the process.
+                int rv = kill(nonChildProcess.Id, SIGKILL);
+                Assert.Equal(0, rv);
+
+                // Wait until the process is reaped.
+                for (int i = 0; i < 100 && rv == 0; i++)
+                {
+                    rv = kill(nonChildProcess.Id, 0);
+                    if (rv == 0)
+                    {
+                        // process still exists, wait some time.
+                        await Task.Delay(100);
+                    }
+                }
+                Assert.NotEqual(0, rv);
+
+                // Call Process.Kill.
+                nonChildProcess.Kill(killTree);
+            }
+
+            Process CreateNonChildProcess()
+            {
+                // Create a process that isn't a direct child.
+                int nonChildPid = -1;
+                RemoteInvokeHandle createNonChildProcess = RemoteInvoke(() =>
+                {
+                    using (RemoteInvokeHandle sleepProcess = RemoteInvoke(RemotelyInvokable.Sleep, RemotelyInvokable.WaitInMS.ToString(),
+                        // Don't pass our standard out to the sleepProcess or the ReadToEnd below won't return.
+                        new RemoteInvokeOptions { StartInfo = new ProcessStartInfo() { RedirectStandardOutput = true }
+                        }))
+                    {
+                        Console.WriteLine(sleepProcess.Process.Id);
```

Comment:
Can we avoid writing to stdout like this?  We try to only do so in error conditions, as otherwise it clutters up test output.

Reply:
Ah, you're redirecting output.  Nevermind.

Reply:
This stdout is redirected and read by the test, it doesn't end up as test output.